### PR TITLE
10207 - Remove withrouter from several components

### PR DESCRIPTION
**Pull request by anjenkin**
**High level description:**

High level description of the PR.

**Technical details:**

Technical details for the knowledge of other developers.

**JIRA Ticket:**
[DEV-10207](https://federal-spending-transparency.atlassian.net/browse/DEV-10207)

The following are ALL required for the PR to be merged:

Author:
- [ ] Linked to this PR in JIRA ticket

Reviewer(s):
- [ ] Design review complete `if applicable`

### CODE REVIEW

File: src/js/components/sharedComponents/sidebar/SidebarLink.jsx [hunk 1]
```diff
@@ -3,69 +3,55 @@
...
+        setUrl(tempUrl);
+    };
 
-        this.setState({
-            url
-        });
-    }
+    useEffect(() => {
+        prepareLink();
+    }, [prepareLink]);
```
Comment (brianpetway) [inline src/js/components/sharedComponents/sidebar/SidebarLink.jsx:31]: This dep isn't needed, I'm assuming it was auto-added, so we should use
// eslint-disable-next-line react-hooks/exhaustive-deps
here

**Comment by brianpetway** [comment]
There are changes to package-lock in this branch.
Were any packages changed for this ticket?

File: src/js/components/sharedComponents/sidebar/SidebarLink.jsx (hunk 1 above)
Comment (anjenkin) [reply src/js/components/sharedComponents/sidebar/SidebarLink.jsx:31]: I'll change it but it doesn't make a difference.

**Comment by anjenkin** [comment]
> There are changes to package-lock in this branch. Were any packages changed for this ticket?

No, there weren't any changes to package.json in the PR, right?

**Comment by brianpetway** [comment]
No, there aren't any changes in package.json but package-lock.json is still in the list of changed files.
But it shouldn't be, should it?

**Comment by anjenkin** [comment]
It should be fine.  Yes, I ran npm install to generate a new package-lock.json.  We should be able to do so on any branch.  If we can't something is wrong with package.json or the local environment.

**Pull request merged**